client/app/house/create/page.tsx

```diff
@@ -1,9 +1,46 @@
 'use client'
 
+import { useEffect, useRef } from 'react'
+import { useRouter, useSearchParams } from 'next/navigation'
+import { Loading } from '@/_components/layout'
+import { Heading } from '@/_components/ui'
+import { INVITED_HOUSE_ID } from '@/_consts'
+import { useGetHouseId } from '@/_hooks'
+import { useCreateUserHouseMutation } from '@/_hooks/api'
+
 export default function HouseCreatePage() {
+  const searchParams = useSearchParams()
+  const invitedHouseId = searchParams.get(INVITED_HOUSE_ID)
+
+  const router = useRouter()
+
+  const hasMutated = useRef(false)
+
+  const { houseId } = useGetHouseId()
+  const { mutate, isPending } = useCreateUserHouseMutation()
+
+  useEffect(() => {
+    if (houseId) {
+      router.push('/')
+    }
+
+    // If user had invitedHouse Id then create user house before showing house setting page
+    if (invitedHouseId && !houseId && !hasMutated.current) {
+      hasMutated.current = true
+      mutate(invitedHouseId)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [houseId, invitedHouseId])
+
+  if (invitedHouseId || isPending) {
+    return <Loading />
```

Comment:
Are you plannning to keep showing the Loading component when the user has invitedHouseId, which is not valid or maybe show an error?

Reply:
No we don't want to keep showing Loading UI! The error handling is in `useCreateUserHouseMutation`!

Reply:
Yeah I get an error toast but I can keep seeing the Loading component when trying to access to the url that has invalid invitedHouseId. So if you plan to show a error component or something, maybe you should show here right?

Reply:
@Atsuki02 
Oh, I get it. Yes, in this case, toast is not enough. Instead of toast, I'll throw an error and show the error component. Thanks! I'll leave a TODO comment or modify it in the upcoming PR before forget about this!